### ISSUE TITLE
net_value limits

### ISSUE THREAD

**Issue by pmargreff** [datasciencebr/serenata-de-amor#135]
Reading [the glossary](http://www.camara.gov.br/cota-parlamentar/2016_Glossario_para_o_CotasNet.pdf) I found some limits to the `subquota_description` so, I tried to find moments this value is bitten.  

The first analysis the max was bitten in R$ 11 M, so I look again and saw the dates the values start to be valid. 

In the second moment following the date the values are validates we can recovered R$ 500 k. I think have made some mistake, I don't believe the value is so large anyway. But, it could get some irregularities, and maybe after the net_value be fixed it could get a few more cases. 

Here is all dates with reference on code: 

[ATO DA MESA Nº 112, DE 15/10/2013](http://www2.camara.leg.br/legin/int/atomes/2013/atodamesa-112-15-outubro-2013-777266-publicacaooriginal-141483-cd-mesa.html)
[ATO DA MESA Nº 4, DE 25/02/2015](http://www2.camara.leg.br/legin/int/atomes/2015/atodamesa-4-25-fevereiro-2015-780188-publicacaooriginal-146197-cd-mesa.html) 
[ATO DA MESA Nº 49 DE 03/09/2015](http://www2.camara.leg.br/legin/int/atomes/2015/atodamesa-49-3-setembro-2015-781493-publicacaooriginal-148023-cd-mesa.html)
[ATO DA MESA Nº 142, DE 28/05/2016](http://imagem.camara.gov.br/Imagem/d/pdf/DCD0020140529SA0790000.PDF#page=3)

**Comment by pmargreff** [comment]
My idea is compare campaign date with the expenses date in publicity, and rewrite and try validate if the `subquota_description` have some coherence with the product/service .

**Comment by anaschwendler** [comment]
Hi @pmargreff!
So, we were starting to apply those rules in a script yesterday, but we find some 
inconsistencies in out dataset (that can be found in: https://github.com/datasciencebr/serenata-de-amor/issues/133) while we were developing it.

By now, we are waiting for a reply from câmara, I think that is better to analyse this before we have the right dataset, what do you think?

**Comment by pmargreff** [comment]
@anaschwendler, ok, thanks for spent some time in. I looked into the pull request made and didn't find anything related with (or I couldn't understand). Anyway, I'll wait the reply to try again.

**Comment by cuducos** [comment]
>  I looked into the pull request made and didn't find anything related with

TL;DR:

* we're not sure if we can take into account rows without `reimbursement_number`.
* maybe the dataset makes more sense with an analysis by receipt not by each line of the dataset (an unique ID would be combination of `applicant_id`, `year` and `document_number`, as we do [to fetch the digitalized image of the receipts](https://github.com/datasciencebr/serenata-de-amor/blob/master/src/fetch_receipts.py#L94-L105)
* @irio and I are still waiting for some responses — check #133 — but we might have a new dataset soon
* if this is the case, we have idea how to handle half of the dataset, i.e. lines without `document_number` (it looks like that before adopting the practice of scanning the receipts, no `document_number` was typed into the system)

**Comment by anaschwendler** [comment]
Hi @pmargreff!
So, as I was talking to you we already developed the new dataset and it's called reimbursement, you can check about in: https://github.com/datasciencebr/serenata-de-amor/pull/134 and https://github.com/datasciencebr/serenata-de-amor/pull/140

Our new dataset script is already running in the serenata-toolbox (you can read more about in https://github.com/datasciencebr/serenata-de-amor/issues/143). Me and @Irio worked in a notebook using this new dataset here: https://github.com/datasciencebr/serenata-de-amor/compare/irio-traveled-speeds (it's a work in progress, and has many things to be done, but you can see how we use the dataset)

Looking of what you've done, me and @cuducos discussed about implement julia to work with our notebooks, because, since all the notebooks should work in anywhere, it would be nice if your script prepare the enviroment to work with julia, what do you think about it?

If there's something that I can help, and if you have any doubts, send me a message at telegram and I can help you! :+1:

**Comment by pmargreff** [comment]
@anaschwendler Hello. 

Nice, I'm glad about that, and I have one question, do you have the csv or xz file in somewhere? I checked into [links file](https://github.com/datasciencebr/serenata-de-amor/blob/master/src/fetch_datasets.py) but haven't found anything. If I need run and build that local, no problem.  

About the the environment, I think it's the way too, but I have no idea how to do that, I can make a script to install the packages inside Julia and IJulia (Jupyter Notebook), but Julia itself is complicated, because the differences between systems and it isn't native in systems like python, I think the easiest way is follow the official site. If you have any suggestion, I'll appreciate.

Maybe in the first time we can start with a [Julia docker container](https://hub.docker.com/_/julia/), how about that @anaschwendler? I never used docker myself, but if someone here have experience I will appreciate a little help (feel free to contact me).

**Comment by cuducos** [comment]
Hi @pmargreff — once more, thanks for you contributions!

>  do you have the csv or xz file in somewhere?

Yes, we're in the middle of a transition: core tasks useful for [Serenata de Amor](https://github.com/datasciencebr/serenata-de-amor) and [Rosie](https://github.com/datasciencebr/rosie) now resides in a specific [Toolbox](https://github.com/datasciencebr/serenata-toolbox). And we still have to [update scripts at this repo to use the toolbox](https://github.com/datasciencebr/serenata-de-amor/issues/143). While we migrate everything you can download the most updated datasets with [this script](https://github.com/datasciencebr/serenata-toolbox/blob/master/serenata_toolbox/datasets.py) and then generate the _reimbursements_ dataset with [this other one](https://github.com/datasciencebr/serenata-toolbox/blob/master/serenata_toolbox/reimbursements.py) (sooner or later it will be downloaded automatically, don't worry). Sorry about the mess, a lot of changes in progress ; )

> but Julia itself is complicated

I don't think it has to be. As we do with Anaconda Python, just point to [their download page](http://julialang.org/downloads/). Maybe @fcevado will be interested in this integration Julia - Jupyter - Serenata (so I tagged him, yay). Does that make sense to you?

> Maybe in the first time we can start with a Julia docker container

Sure thing Julia should be ready to use in Docker and non-Docker installation. If you need help on that, feel free to open an issue and ask for help from people who have been working with Docker here (e.g. @dsakuma @gwmoura for example). I guess it's not a best practice, but installing it [via `apt-get`](http://julialang.org/downloads/platform.html) might work in the container as it's a Ubuntu-based image.

**Comment by fcevado** [comment]
I was reading the whole issue, as I see there are those three points:
1. Integrate julia code with `serenata-toolbox`
2. Automate the julia environment setting
3. Prepare a Docker instalation

If I'm correct to assume that, I can suggest somethings:
1. There are some tools to use python with julia(i.e. [JuliaPy](https://github.com/JuliaPy)). I guess the only problem would be installing the package manually, since `serenata-toolbox` isn't published.
2. I don't know any way to automate package installation in julia(as far as I know you must do it in the beggining of every script). :disappointed:  
3. There is a [Julia Image](https://hub.docker.com/r/julialang/julia/) with `IJulia` and `Jupiter` included. Sadly it's based on `ubuntu` not on `alpine`, probably it's a big image.

I can try to help with any of those points(if I'm assuming them correctly).

**Comment by dsakuma** [comment]
Hi @pmargreff! The docker image that we are using already contains Julia. You just need to select the Julia kernel in notebook.

**Comment by cuducos** [comment]
Just a minor correction, @fcevado: Julia should be available here at `serenata-de-amor`, no need for it in `serenata-toolbox`, ok?

**Comment by cuducos** [comment]
@pmargreff the reimbursements dataset is in [`datasets.py`](https://github.com/datasciencebr/serenata-toolbox/blob/master/serenata_toolbox/datasets.py) now ; )

You can install it with: `$ pip install -e git+https://github.com/datasciencebr/serenata-toolbox.git#egg=serenata-toolbox` and then this will doo the trick:

```python
from serenata_toolbox import fetch_latest_backup
fetch_latest_backup('data/')
```
